sink/sink

```diff
@@ -89,9 +97,31 @@ def expand(val, env):
     else:
         return val
 
+# Compare two JSON style values to see if one is contained in the other
+# For lists and dict make sure all items in expect are also present in result
+# For any other values a strict comparison is used
+def contains(expect, result):
+    if isinstance(expect, list) and isinstance(result, list):
+        for i in range(0, len(expect)):
+            res = result[i] if i < len(result) else None
+            if not contains(expect[i], res):
+                return False
+        return True
```

Comment:
Ah, so this fourth commit already (attempts to) change that to a real "contains". It still seems a bit weird to me that multiple occurrences count as "false", though? What's the use case for that?

With this algorithm you lose the ordering of `expect` -- i. e. `contains([2, 1], [1, 2, 3])` should now be `True`.
(There are a [couple of recipes on StackExchange]http://stackoverflow.com/questions/3847386/testing-if-a-list-contains-another-list-with-python) for this, BTW). However, I think in this GitHub case this "is a subset" behaviour is actually desirable. You can write that much more succinctly as `if set(expect).issubset(result)`.